percona/controller/pgcluster/controller_test.go

```diff
@@ -1622,3 +1622,118 @@ var _ = Describe("Validate TLS", Ordered, func() {
 		checkSecretProjectionWithCA(cr, cr.Spec.Secrets.CustomReplicationClientTLSSecret, secretName)
 	})
 })
+
+type saTestClient struct {
+	client.Client
+
+	crName string
+	ns     string
+}
+
+func (sc *saTestClient) checkObject(ctx context.Context, obj client.Object) error {
+	sts, ok := obj.(*appsv1.StatefulSet)
+	if !ok {
+		return nil
+	}
+	serviceAccountName := sts.Spec.Template.Spec.ServiceAccountName
+	if serviceAccountName == "" {
+		panic("it's not expected to have empty service account name")
```

Comment:
Since `checkObject(...)` returns an error and when used we do this

```
	if err := sc.checkObject(ctx, obj); err != nil {
		panic(err) // should panic because reconciler can ignore the error
	}
```

I think this check can return an error and let the caller panic it. WDYT?

Reply:
https://github.com/percona/percona-postgresql-operator/pull/1125/commits/f1f4ce304e4a2d82436d5acc12d6022ae0db7fc8